xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/internal/email/DefaultNotificationEmailUserPreferenceManager.java

```diff
@@ -77,49 +84,67 @@ public class DefaultNotificationEmailUserPreferenceManager implements Notificati
     @Override
     public NotificationEmailDiffType getDiffType()
     {
-        return getDiffType(documentAccessBridge.getCurrentUserReference());
+        return getDiffType(CurrentUserReference.INSTANCE);
+    }
+
+    @Override
+    public NotificationEmailDiffType getDiffType(UserReference userReference)
+    {
+        return getStaticListPropertyPreference(DIFF_TYPE, NotificationEmailDiffType.class,
+            NotificationEmailDiffType.STANDARD, userReference);
     }
 
     @Override
-    public NotificationEmailDiffType getDiffType(String userId)
+    public NotificationEmailInterval getInterval()
     {
-        DocumentReference user = referenceResolver.resolve(userId);
-        return getDiffType(user);
+        return getInterval(CurrentUserReference.INSTANCE);
     }
 
-    private NotificationEmailDiffType getDiffType(DocumentReference user)
+    @Override
+    public NotificationEmailInterval getInterval(UserReference userReference)
+    {
+        return getStaticListPropertyPreference(INTERVAL, NotificationEmailInterval.class,
+            NotificationEmailInterval.DAILY, userReference);
+    }
+
+    private <T extends Enum<T>> T getStaticListPropertyPreference(String propertyName,
+        Class<T> propertyEnum, T propertyDefaultValue, UserReference user)
     {
         try {
+
+            DocumentReference userDocumentReference = documentUserSerializer.serialize(user);
+
             // Get the config of the user
             DocumentReference emailClassReference = new DocumentReference(EMAIL_PREFERENCES_CLASS,
-                    user.getWikiReference());
-            Object diffType = documentAccessBridge.getProperty(user, emailClassReference, DIFF_TYPE);
-            if (diffType != null && StringUtils.isNotBlank((String) diffType)) {
-                return NotificationEmailDiffType.valueOf((String) diffType);
+                userDocumentReference.getWikiReference());
+            Object value = documentAccessBridge.getProperty(userDocumentReference, emailClassReference, propertyName);
+            if (value != null && StringUtils.isNotBlank((String) value)) {
+                return Enum.valueOf(propertyEnum, ((String) value).toUpperCase());
```

Comment:
Minor comment but would be cleaner to have a single return value probably.

Reply:
Thanks but not sure what you mean here. Do you mean introducing `T returnValue = propertyDefaultValue;` at the beginning, and reassigning it in `if {...} else if {...} else {...}` blocks with just a final `return` statement, is that right?

Reply:
Yes that's what I mean.

Reply:
Note that it's not something we documented as best practice to my knowledge but we generally try to adopt this style when possible.

Reply:
ok thx done this